File: lib/svg/tests/tspanCheck.py
from svg import SVG, Text, Tspan

valid = SVG(100, 100).setData([Text().setData(["hello", Tspan().setData(["nested"])])])

invalidList = [
    SVG(100, 100).setData([Tspan()]),
    SVG(100, 100).setData([Tspan().setData(["ouho", Tspan().setData(["inner"])])]),
]

assert valid.checkTspan()

for invalid in invalidList:
    assert not invalid.checkTspan()

try:
    invalidList[0].export()
    assert False
except SyntaxError as err:
    assert (
        f"{err}" == "Tspan instances must be children of other Tspan or Text instances!"
    )

File: lib/svg/tests/tspanCheck.cc
#include <cassert>
#include <iostream>

#include "../elements/shapes/text.hh"
#include "../svg.hh"

int main(void)
{
    Tspan t2{};
    t2.setData({ "nested" });
    Text t1{};
    t1.setData({ "hello", t2 });
    SVG valid{ 100, 100 };
    valid.setData({ t1 });

    SVG invalid1{ 100, 100 };
    invalid1.setData({ Tspan{} });

    Tspan t3{};
    t3.setData({ "inner" });
    SVG invalid2{ 100, 100 };
    invalid2.setData({ "wrong", t3 });

    assert(valid.checkTspan());
    assert(!invalid1.checkTspan());
    assert(!invalid2.checkTspan());

    try
    {
        std::cout << invalid1;
        assert(false);
    }
    catch (const std::logic_error& e)
    {
        assert(std::string{ e.what() }
               == "SVG: print_: Tspan instances must be children of other "
                  "Tspan or Text instances!");
    }
}